3rdparty/indi-astromechfoc/astromech_focuser.cpp

```diff
@@ -0,0 +1,314 @@
+ *
+************************************************************************************/
+astromechanics_foc::astromechanics_foc()
+{
+    setVersion(0, 1);
+    FI::SetCapability(FOCUSER_CAN_ABS_MOVE | FOCUSER_CAN_REL_MOVE);
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+const char *astromechanics_foc::getDefaultName()
+{
+    return (const char *)"Astromechanics FOC";
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+void astromechanics_foc::ISGetProperties(const char *dev)
+{
+    if (dev != nullptr && strcmp(dev, getDeviceName()) != 0)
+        return;
+
+    INDI::Focuser::ISGetProperties(dev);
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+bool astromechanics_foc::initProperties()
+{
+    INDI::Focuser::initProperties();
+
+    FocusMaxPosN[0].min = FOC_POSMIN_HARDWARE;
+    FocusMaxPosN[0].max = FOC_POSMAX_HARDWARE;
+    FocusMaxPosN[0].step = (FocusMaxPosN[0].max - FocusMaxPosN[0].min) / 20.0;
+    FocusMaxPosN[0].value = 5000;
+    FocusAbsPosN[0].max = FocusAbsPosN[0].value;
+
+    FocusRelPosN[0].min = 1;
+    FocusRelPosN[0].max = FocusAbsPosN[0].max;
+    FocusRelPosN[0].value = 500;
+    FocusRelPosN[0].step = 10;
+
+    IUFillNumber(&AppertureN[0], "LENS_APP", "Index", "%2d", 0, 22, 1, 0);
+    IUFillNumberVector(&AppertureNP, AppertureN, 1, getDeviceName(),"LENS_APP_SETTING", "Apperture", MAIN_CONTROL_TAB, IP_RW, 60, IPS_IDLE);
+    serialConnection->setDefaultBaudRate(Connection::Serial::B_38400);
+    return true;
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+bool astromechanics_foc::updateProperties()
+{
+    // Get Initial Position before we define it in the INDI::Focuser class
+    FocusAbsPosN[0].value = GetAbsFocuserPosition();
+
+    INDI::Focuser::updateProperties();
+
+    if (isConnected())
+    {
+        defineSwitch(&StartSavedPositionSP);
+        defineNumber(&AppertureNP);
+    }
+    else
+    {
+        deleteProperty(StartSavedPositionSP.name);
+        deleteProperty(AppertureNP.name);
+    }
+
+    return true;
+}
+
+/************************************************************************************
+ *
+************************************************************************************/
+bool astromechanics_foc::Handshake()
+{
+    char FOC_cmd[32] = "P#";
+    char FOC_res[32] = {0};
+    int FOC_pos_measd = 0;
+    int nbytes_written = 0;
+    int nbytes_read = 0;
+
+    LOGF_INFO("Handshake", 0);
+    if (CanAbsMove()) {
+        LOGF_INFO("CanAbsMove",0);
```

Comment:
No need to pass 0, you can just say LOG_INFO("CanAbsMove")

But this is for the user, so better have a better message like LOG_INFO("Focuser supports absolute motion");

Reply:
Removed that - was for development purposes only.